### ISSUE TITLE
RuntimeError: deque mutated during iteration

### ISSUE THREAD

**Issue by talfirevic** [SerpentAI/D3DShot#8]
```
import d3dshot
import time

d = d3dshot.create(capture_output="numpy")
d.capture(region=(100,100,300,300), target_fps=30)
time.sleep(20)
d.stop()
d.frame_buffer_to_disk()
```

dumping frame buffer to disk always yields following error:
`File "C:\ProgramData\Anaconda3\envs\testing_env\lib\site-packages\d3dshot\d3dshot.py", line 119, in frame_buffer_to_disk
    for i, frame in enumerate(self.frame_buffer):
RuntimeError: deque mutated during iteration`

**Comment by nbrochu** [comment]
![image](https://user-images.githubusercontent.com/35039/83115283-2554a980-a098-11ea-825e-300e337758f4.png)

Runs fine here on 3.8.3. It's hard to tell why you are getting this but going by the error message: `d.stop()` might not be done running or getting ignored. `d.frame_buffer_to_disk()` iterates over what's in the frame buffer to save the individual images but if it receives new data while it does that, Python is not going to like it.

I'll look into why that could be happening at all and to see if there is more reliable way to stop the capture. I could also just copy the frame buffer to a separate iterable only for the duration of the image saving on disk.

Thanks for reporting this.

**Comment by nbrochu** [comment]
This should be fixed/improved in 0.1.4
